### DISCUSSION TITLE
When will Homebrew stop supporting Intel Macs?

### DISCUSSION THREAD

**Discussion by rodrigc** [Homebrew/discussions#2607] [Everyday usage] [answered]
I am thinking of buying a new Macbook Pro, but I still want to buy a 2019 i9 Intel Mac, because I use Virtualbox, vagrant, and Docker quite heavily for cloud work, and all that stuff works better on Intel than M1.

I use Homebrew quite heavily on Mac.
When do you think Homebrew will stop supporting Intel?

On a separate note, I contacted https://support.apple.com and asked them when they will stop providing macOS updates
and software support for Intel.  Apple support told me that they did not know, because there is no official statement about this from Apple.

If Homebrew can support Intel for the next 5-7 years, that is probably good enough for me.

**Comment by SMillerDev** [comment, marked as answer]
We'll probably support Intel until Apple has no supported OS version left for Intel.

I realize that doesn't help you, but I can't really promise support for software beyond Apple's support.

**Reply by Bo98** [reply]
Given we support the last 3 OS versions, this will effectively be +2 years from the release date of the first macOS version without Intel support. Given Apple are still selling Intel Macs, I don't anticipate a new macOS version without Intel support is imminent.

<Insert standard disclaimer that we don't guarantee anything and we're not responsible for financial decisions of users.>

**Comment by rodrigc** [comment]
@SMillerDev @Bo98  Thanks for your responses.  Your answers seem quite reasonable.  Since Apple has not formally announced a timetable for when they will drop Intel support, that seems the best answer we can give for right now.

In future, as Apple announces their deprecation schedule for Intel, we can update this thread.

On a side note, I have a 2013 Macbook Pro which works only with Big Sur, and is not on the list of supported
hardware for Monterrey: [macOS Monterey is compatible with these computers](https://support.apple.com/en-us/HT212551).

So I guess over time, Apple drops support for older hardware in new OS releases, and that will be their strategy for
dropping Intel support, but it may take some time (years).

**Comment by abdul** [comment]
Apple is still supporting Macbook Intel. What is sad that, Homebrew has decided to stop supporting macOS (15.7.7) Intel (
MacBook Pro "Core i7" 2.6 16" 2019). Following is what I have started seeing since yesterday (Sept 1, 2026):

<img width="823" height="213" alt="image" src="https://github.com/user-attachments/assets/5402f705-d85f-4945-9f7d-8d4a0af79b62" />

**Reply by SMillerDev** [reply]
> Apple is still supporting Macbook Intel.

For another couple days that is true. When 27 is out, Apple will only have an OS that they occasionally patch some of the security holes for that runs on Intel processors.

https://github.com/orgs/Homebrew/discussions/7044#discussioncomment-18231338 has more details

**Reply by abdul** [reply]
Yes, that is true. Homebrew is not Apple; Homebrew should support Mac Intel as long as they can -- Github wouldn't drop intel support this year, there is time to it. Why did Homebrew stop providing bottles for intel? It's painful to have all formulas getting built on local machine. It's not easy for someone, user since 2009-2010,  to migrate to MacPorts or alternative with hundreds of formulas, and an entire ecosystem built around it. I am upset :-(

**Reply by SMillerDev** [reply]
> Homebrew should support Mac Intel as long as they can

So, the giant corporation is okay to drop support but the volunteers should keep volunteering their time indefinitely? That seems kind of unfair.

> Why did Homebrew stop providing bottles for intel?

That's explained in the linked post. But nothing is stopping you from setting up a new tap that builds bottles for intel using GitHub actions.

**Reply by Terre6** [reply]
Apple is a hardware company, and it's in their commercial interest to make older machines obsolete in order to sell new ones. I agree with @abdul that Homebrew doesn't need to follow that path. Projects like [OpenCore Legacy Patcher](https://github.com/dortania/Opencore-Legacy-Patcher) (also run by volunteers) show that older Macs are still perfectly capable.

That said, there's no use crying over spilled milk: I understand the decision won't be reversed, and I respect the time volunteers put into this project. It's with a heavy heart that I spent today uninstalling Homebrew and migrating to MacPorts.

MacPorts supports even the oldest macOS versions, and many formulae have the same name as the corresponding ports. For the others, there is a [Search ports](https://ports.macports.org/search/) page; just uncheck "Only match by port names" so that descriptions are included too (e.g. `jpeg-xl` is `libjxl`).

By the way, MacPorts has "variants", which work much like the old `homebrew-core` options. Remember when you could simply add `--with-foo` or `--without-bar` to a formula? Those were the days. I have extremely fond memories of that golden age of Homebrew, where the default configuration wasn't a requirement, and users had choice.

**Reply by bennett-elder** [reply]
> > Homebrew should support Mac Intel as long as they can
> 
> So, the giant corporation is okay to drop support but the volunteers should keep volunteering their time indefinitely? That seems kind of unfair.
> 

That misrepresents the situation. My mac has another two years of security updates. Homebrew is dropping support the moment Apple finally put a concrete end date on their Intel phase out.

**Reply by p-linnane** [reply]
Apple explicitly warns that [older macOS versions do not receive fixes for all known security issues](https://support.apple.com/guide/deployment/about-software-updates-depc4c80847a/web). A trillion-dollar company providing some security updates doesn’t provide Homebrew’s volunteers with the time or resources to keep supporting those systems.

As for the “golden age”: we’re in it. We have more users than ever, our maintainers are happy, and we were able to rebuild almost 7,000 packages between the macOS 27 RC release on September 9 and the final release on September 14, precisely because Intel was no longer slowing down our build infrastructure. These decisions have tangible benefits for the project and its users.

**Reply by bennett-elder** [reply]
I read that to mean "if a security fix applies to a feature that only works on Apple Silicon like Apple Intelligence, only users on that hardware will see a security update" not "sometimes we don't patch issues we know about in macOS 15".

**Reply by p-linnane** [reply]
That’s an incorrect reading. Incomplete security patching of older macOS releases has been documented for years.

For example, Apple fixed CVE-2025-31250 in Sequoia 15.5, but the researcher reproduced it on Sonoma 14.7.6, released the same day. He then reports that Apple Product Security explicitly confirmed Sonoma and Ventura would not receive a fix. This affects macOS’s existing permission system. Here's a link to the [researcher’s write-up, including Apple’s response](https://wts.dev/posts/tcc-who/).

**Reply by bennett-elder** [reply]
yikes. while inconvenient to my point, I appreciate the correction.

**Reply by slayernsocal** [reply]
so how does one go about using Macports or any program that is already based in hombrew casks port over to being used  in Macports. The link you provide within the notification Is great, yet Homebrew provides absolutely no following information on what do to in handling a transition to MacPorts etc. Yes I have read the MacPorts website up and down and it's rather daunting and very unhelpful in trying to use it once it's installed. So how do I run my program I was using daily on. t his new platform.

**Reply by Terre6** [reply]
@slayernsocal Just a heads-up that I'm not a Homebrew or MacPorts member, just a fellow user who went through this.

Before starting, the one thing worth doing first is checking whether the essential things you use are available as ports. Search <https://ports.macports.org/search/> to know upfront whether the move is worth it for you.

Below are the exact steps that worked for me.

**1) Save the list of everything installed with Homebrew**

```sh
# Keep a record
brew bundle dump --file="$HOME/Downloads/brew-backup.txt"
```

This .txt file was my checklist for MacPorts.

**2) Uninstall Homebrew entirely**

```sh
# Stop the services
brew services stop --all
# Remove all Homebrew packages
brew list --formula | xargs brew uninstall --force --ignore-dependencies
brew list --cask | xargs brew uninstall --cask --force
# Uninstall Homebrew itself
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

**3) Install MacPorts**

Run the `.pkg` for your macOS version from <https://www.macports.org/install.php>

**4) Open a new terminal window, then check the install**

```sh
port version
port diagnose
```

**5) Find and install the ports**

```sh
# Search for ports whose name contains "jxl"
port search --name jxl
# Install the one you want, e.g. "libjxl"
sudo port -N install libjxl
```

**Reply by p-linnane** [reply]
> The link you provide within the notification Is great, yet Homebrew provides absolutely no following information on what do to in handling a transition to MacPorts etc.

Because we are not MacPorts and don't know their processes. This is our best effort for users on Intel, not a comprehensive guide.
